lib/views/pages/login_signup/login_page.dart

```diff
@@ -367,15 +371,48 @@ class _LoginScreenState extends State<LoginPage> with TickerProviderStateMixin {
                                 RaisedButton(
                                     padding: EdgeInsets.all(12.0),
                                     shape: StadiumBorder(),
-                                    child: Text(
-                                      saveMsg,
-                                    ),
+                                    child: isUrlCalled
+                                        ? SizedBox(
+                                            height: 14,
+                                            width: 14,
+                                            child: CircularProgressIndicator(
+                                              backgroundColor: Colors.white,
+                                            ),
+                                          )
+                                        : Text(
+                                            saveMsg,
+                                          ),
                                     color: Colors.white,
                                     onPressed: () async {
                                       if (_formKey.currentState.validate()) {
                                         _formKey.currentState.save();
-                                        setAPIURL();
-                                        _setURL();
+
+                                        setState(() {
+                                          isUrlCalled = true;
+                                        });
+
+                                        try {
+                                          final response = await http.get(
```

Comment:
I would suggest handling this `async `task in a separate function for making the code testable. But, wait for the mentor's call also. They would surely have a better approach to this.